packages/app-headless-cms/src/admin/components/ContentEntries/Filters/RefFieldRenderer/domain/EntryRepositoryFactory.ts

```diff
@@ -0,0 +1,28 @@
+import { ApolloClient } from "apollo-client";
+import { EntriesGatewayInterface, EntriesGraphQLGateway } from "../adapters";
+import { EntryRepository } from "./EntryRepository";
+
+export class EntryRepositoryFactory {
+    private gateway: EntriesGatewayInterface | undefined;
+    private cache: Map<string, EntryRepository> = new Map();
+
+    getRepository(client: ApolloClient<any>, modelIds: string[]) {
```

Comment:
It would be cleaner if you passed the Gateway instance instead of the `client`. Because if you look closely, you're passing a `client` just so you can pass it further down to the Gateway constructor, and nothing else. So it's prop drilling (or dependency drilling), which should be avoided. Pass a gateway, and modelIds to the factory.